android/src/main/java/com/oney/WebRTCModule/DailyWebRTCDevicesManager.java

```diff
@@ -0,0 +1,201 @@
+    }
+
+    public void startMediaDevicesEventMonitor(){
+        this.audioManager.registerAudioDeviceCallback(audioDeviceCallback, null);
+    }
+
+    private CameraEnumerator createCameraEnumerator() {
+        boolean camera2supported = false;
+        try {
+            camera2supported = Camera2Enumerator.isSupported(this.reactContext);
+        } catch (Throwable tr) {
+            // Some devices will crash here with: Fatal Exception: java.lang.AssertionError: Supported FPS ranges cannot be null.
+            // Make sure we don't.
+            Log.w(TAG, "Error checking for Camera2 API support.", tr);
+        }
+        CameraEnumerator cameraEnumerator = null;
+        if (camera2supported) {
+            Log.d(TAG, "Creating video capturer using Camera2 API.");
+            cameraEnumerator = new Camera2Enumerator(this.reactContext);
+        } else {
+            Log.d(TAG, "Creating video capturer using Camera1 API.");
+            cameraEnumerator = new Camera1Enumerator(false);
+        }
+        return cameraEnumerator;
+    }
+
+    ReadableArray enumerateDevices() {
+        WritableArray devicesArray = Arguments.createArray();
+        this.fillVideoInputDevices(devicesArray);
+        this.fillAudioDevices(devicesArray);
+        return devicesArray;
+    }
+
+    private void fillVideoInputDevices(WritableArray enumerateDevicesArray){
+        String[] devices = cameraEnumerator.getDeviceNames();
+        for (int i = 0; i < devices.length; ++i) {
+            String deviceName = devices[i];
+            boolean isFrontFacing;
+            try {
+                // This can throw an exception when using the Camera 1 API.
+                isFrontFacing = cameraEnumerator.isFrontFacing(deviceName);
+            } catch (Exception e) {
+                Log.e(TAG, "Failed to check the facing mode of camera");
+                continue;
+            }
+            String label = isFrontFacing ? "Front camera" : "Rear camera";
+            String deviceID = isFrontFacing ? "CAMERA_USER" : "CAMERA_ENVIRONMENT";
+            WritableMap params = this.createWritableMap(deviceID, label, DeviceKind.VIDEO_INPUT.getKind());
+            params.putString("facing", isFrontFacing ? "user" : "environment");
+            enumerateDevicesArray.pushMap(params);
+        }
+    }
+
+    private void fillAudioDevices(WritableArray enumerateDevicesArray){
+        AudioDeviceInfo[] audioOutputDevices = this.audioManager.getDevices(AudioManager.GET_DEVICES_OUTPUTS);
+
+        boolean isWiredHeadsetPlugged = Arrays.stream(audioOutputDevices).anyMatch(
+                device -> device.getType() == AudioDeviceInfo.TYPE_WIRED_HEADSET ||
+                        device.getType() == AudioDeviceInfo.TYPE_WIRED_HEADPHONES
+        );
+
+        WritableMap params = isWiredHeadsetPlugged ?
+                this.createWritableMap(AudioDeviceType.WIRED_OR_EARPIECE.toString(), "Wired headset", DeviceKind.AUDIO.getKind()) :
```

Comment:
Just a reminder to ourselves to mention this when we're writing documentation for this new functionality: our device label will say "Wired headset" even when technically it's just a wired _headphone_ with a microphone